public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -467,7 +467,7 @@ export function processRangeQueryResponse(
   switch (response.data.resultType) {
     case LokiResultType.Stream:
       return of({
-        data: lokiStreamsToDataframes(response.data.result, target, limit, config, reverse),
+        data: lokiStreamsToDataframes(limit > 0 ? response.data.result : [], target, limit, config, reverse),
```

Comment:
Gonna change it with an implementation of new form styles - with `limit == 0` request isn't going to be sent, we'll just clear the result of this query and display a validation error.